src/main/scala/no/ndla/validation/TagValidator.scala

```diff
@@ -69,46 +72,50 @@ class TagValidator {
     val validationErrors = attributesAreLegal(fieldName, allAttributesOnTag, ResourceHtmlEmbedTag) ++
       attributesContainsNoHtml(fieldName, legalAttributes) ++
       verifyAttributeResource(fieldName, legalAttributes) ++
-      verifyParent(fieldName, legalAttributes, embed) ++
+      verifyParent(fieldName, legalAttributes, embed, validateParent) ++
       verifyRequiredOptional(fieldName, legalAttributes, embed)
 
     validationErrors
   }
 
   private def verifyParent(fieldName: String,
                            attributes: Map[TagAttributes.Value, String],
-                           embed: Element): Seq[ValidationMessage] = {
-    attributes
-      .get(TagAttributes.DataResource)
-      .map(resourceTypeString => {
-        ResourceType.valueOf(resourceTypeString) match {
-          case Some(resourceType) =>
-            val attributeRulesForTag = EmbedTagRules.attributesForResourceType(resourceType)
-            val legalOptionals = attributeRulesForTag.optional.flatten.toSet
-            val legalOptionalAttributesUsed = attributes.keySet.intersect(legalOptionals)
-
-            attributeRulesForTag.mustBeDirectChildOf.toSeq.flatMap(parentRule => {
-              val parent = embed.parent()
-              val expectedButMissingParentAttributes = parentRule.requiredAttr.filterNot {
-                case (attrKey, attrVal) => parent.attr(attrKey) == attrVal
-              }
-
-              if (parent.tagName() != parentRule.name || expectedButMissingParentAttributes.nonEmpty) {
-                val messageString =
-                  s"Embed tag with '${resourceType.toString}' requires a parent '${parentRule.name}', with attributes: '${parentRule.requiredAttr
-                    .map { case (key, value) => s"""$key="$value"""" }
-                    .mkString(", ")}'"
-
-                Seq(ValidationMessage(fieldName, messageString))
-              } else {
-                Seq.empty
-              }
-            })
-          case _ =>
-            Seq(ValidationMessage(fieldName, "Something went wrong when determining resourceType of embed"))
-        }
-      })
-      .getOrElse(Seq.empty)
+                           embed: Element,
+                           validateParent: Boolean = true): Seq[ValidationMessage] = {
```

Comment:
Er det ikke egentlig slik at parent bare valideres unntaksvis? Om så bør kanskje default være `false`? 🐛

Reply:
Uhm, vi vil vel gjerne validere parent'en (dersom det er en regel om det) i de fleste tilfeller?
Valideringen vil jo kun skje dersom det finnes en regel for å validere parenten.